tests/timeago.test.js

```diff
@@ -0,0 +1,130 @@
+var should = require('should');
+var levels = require('../lib/levels');
+var times = require('../lib/times');
+
+describe('timeago', function() {
+  var ctx = {};
+  ctx.ddata = require('../lib/data/ddata')();
+  ctx.notifications = require('../lib/notifications')(env, ctx);
+  ctx.language = require('../lib/language')();
+  ctx.settings = require('../lib/settings')();
+  ctx.settings.heartbeat = 0.5; // short heartbeat to speedup tests
+
+  var timeago = require('../lib/plugins/timeago')(ctx);
+
+  var env = require('../env')();
+
+  function freshSBX () {
+    //set extendedSettings right before calling withExtendedSettings, there's some strange test interference here
+    env.extendedSettings = { timeago: { enableAlerts: true } };
+    var sbx = require('../lib/sandbox')().serverInit(env, ctx).withExtendedSettings(timeago);
+    return sbx;
+  }
+
+  it('Not trigger an alarm when data is current', function(done) {
+    ctx.notifications.initRequests();
+    ctx.ddata.sgvs = [{ mills: Date.now(), mgdl: 100, type: 'sgv' }];
+
+    var sbx = freshSBX();
+    timeago.checkNotifications(sbx);
+    should.not.exist(ctx.notifications.findHighestAlarm('Time Ago'));
+
+    done();
+  });
+
+  it('Not trigger an alarm with future data', function(done) {
+    ctx.notifications.initRequests();
+    ctx.ddata.sgvs = [{ mills: Date.now() + times.mins(15).msecs, mgdl: 100, type: 'sgv' }];
+
+    var sbx = freshSBX();
+    timeago.checkNotifications(sbx);
+    should.not.exist(ctx.notifications.findHighestAlarm('Time Ago'));
+
+    done();
+  });
+
+
+  it('should trigger a warning when data older than 15m', function(done) {
+    ctx.notifications.initRequests();
+    ctx.ddata.sgvs = [{ mills: Date.now() - times.mins(16).msecs, mgdl: 100, type: 'sgv' }];
+
+    var sbx = freshSBX();
+    timeago.checkNotifications(sbx);
+
+    var currentTime = new Date().getTime();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['currentTime' is assigned a value but never used.](https://app.codacy.com/manual/m-kozak/cgm-remote-monitor/pullRequest?prid=1196151)